### PR TITLE
feat: merge alloy-rs/ruint (ruint2)

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Internal changes:
- move crates to crates/*
- improve uint features and deps
  - remove build.rs feature "detection" by unifying cfgs into `feature = "nightly"`: https://github.com/recmo/uint/blob/35167fbf0406c8f925608225225f91f5612e371b/build.rs#L4C8-L4C76
- move benches to benches/*, move criterion to dev deps
- `git rm Cargo.lock`
- update all dependencies to latest versions
- move security audits to cargo-deny
- remove profile overrides (#247)
- remove "dyn" stuff, since it's wholly incomplete
- bumps `ruint` to `1.9.0`, and `ruint-macro` to `1.1.0`

Public changes listed in [`CHANGELOG.md`](https://github.com/alloy-rs/uint/blob/dani/sync/CHANGELOG.md)

Dependabot:
- Closes #221
- Closes #251
- Closes #252
- Closes #258
- Closes #259

Other:
- Closes #247

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [x] Updated the changelog

### CODE REVIEW

File: .github/scripts/issue_tracker.py [hunk 1]
```diff
@@ -3,45 +3,46 @@
 # TODO: The git blame is wrong. It seems to take the last commit instead.
```
Comment (prestwich) [inline .github/scripts/issue_tracker.py:3]: can we restore this file? looks like only cosmetic changes and i don't care about python formatting

File: .editorconfig [hunk 2]
```diff
@@ -8,6 +11,4 @@ trim_trailing_whitespace = true
 insert_final_newline = true
 indent_style = space
 indent_size = 4
-
-[*.{yml,yaml}]
```
Comment (prestwich) [inline .editorconfig:12]: what's the reason for changing this?

File: .github/workflows/audit.yml [hunk 3]
```diff
@@ -1,25 +0,0 @@
-name: Security Audit
```
[the remark this answers is not in the document]
Comment (DaniPopes) [reply .github/workflows/audit.yml:1]: replaced with cargo deny. pls read pr description

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,128 +1,118 @@
 name: CI
 
-on: [push, pull_request]
+on:
```
[the remark this answers is not in the document]
Comment (DaniPopes) [reply .github/workflows/ci.yml:3]: basically alloy-core workflow + kept coverage workflow

File: .editorconfig (hunk 2 above)
Comment (DaniPopes) [reply .editorconfig:12]: same as everywhere else

Comment (prestwich) [reply .editorconfig:12]: what's everywhere?

Comment (DaniPopes) [reply .editorconfig:12]: Why does formatting matter so much :smile: ?

https://github.com/alloy-rs/core/tree/main/.github/workflows
https://github.com/gakonst/ethers-rs/tree/master/.github/workflows
https://github.com/foundry-rs/foundry/tree/master/.github/workflows

**Review by prestwich — CHANGES_REQUESTED** [review]
The scope of this PR is too broad for us to merge as temporary maintainers. First, I'm unwilling to overwrite repo conventions with our own opinions. We're stewards, not owners here. Second, there's no effective record of the changes being made, as all changes are in a single PR, with functional and non-functional changes mixed into the individual commits.

Next steps should be:

1. Remove non-functional changes:
- Formatter settings
- `.github` folder changes
- File renames
- repo directory structure
- etc

2. Open issues for each functional change. Each change should be a single PR with an associated issue

**Comment by DaniPopes** [comment]
Closed in favor of #262, #263, and #264

**Pull request closed without merging**